eva_cttv_pipeline/trait_mapping/ols.py

```diff
@@ -44,13 +31,26 @@ def get_ontology_label_from_ols(ontology_uri: str) -> str:
     :return: Term label for the ontology uri provided in the parameters.
     """
     url = build_ols_query(ontology_uri)
-    label = request_retry_helper(get_label_from_ols, 4, url)
-    return label
-
-
-def build_ols_query(ontology_uri: str) -> str:
-    """Build a url to query OLS for a given ontology uri."""
-    return "https://www.ebi.ac.uk/ols/api/terms?iri={}".format(ontology_uri)
+    json_response = request_retry_helper(url)
+
+    if not json_response:
+        logger.warning('Failed to query OLS for URL {}'.format(url))
+        return None
+
+    # If the '_embedded' section is missing from the response, it means that the term is not found in OLS
+    if '_embedded' not in json_response:
+        logger.warning('OLS queried OK but did not return any results for URL {}'.format(url))
+        return None
+
+    # Go through all terms found by the requested identifier and try to find the one where the _identifier_ and the
+    # _term_ come from the same ontology (marked by a special flag). Example of such a situation would be a MONDO term
+    # in the MONDO ontology. Example of a reverse situation is a MONDO term in EFO ontology (being *imported* into it
+    # at some point).
+    for term in json_response["_embedded"]["terms"]:
+        if term["is_defining_ontology"]:
+            return term["label"]
+
+    logger.warning('OLS queried OK, but there is no defining ontology in its results for URL {}'.format(url))
```

Comment:
I assume that this function ending without an explicit return will give a `None` if the caller tries to assign it, but consider making this explicit.
```suggestion
    logger.warning('OLS queried OK, but there is no defining ontology in its results for URL {}'.format(url))
    return None
```

Reply:
Good point, fixed in https://github.com/EBIvariation/eva-cttv-pipeline/pull/76/commits/919df5b41acc3aa12c1e21a290ceeaf2bad90e6f